src/lib.rs

```diff
@@ -679,7 +703,7 @@ pub mod tests {
             } else {
                 0
             };
-            let poly = Polynomial::rand(degree, rng);
+            let poly = Polynomial::rand(degree * if segmented { 10 } else { 1 }, rng);
```

Comment:
I would choose a non-integer multiple of `degree`.

Reply:
The unit-test function is changed for the segmented test